pennylane/transforms/dynamic_one_shot.py

```diff
@@ -266,18 +276,28 @@ def measurement_with_no_shots(measurement):
             )
         if interface != "jax" and m.mv and not has_valid:
             meas = measurement_with_no_shots(m)
+        elif m.mv and active_jit:
+            meas = gather_mcm_jit(m, mcm_samples, is_valid)
         elif m.mv:
             meas = gather_mcm(m, mcm_samples, is_valid)
         elif interface != "jax" and not has_valid:
             meas = measurement_with_no_shots(m)
             m_count += 1
         else:
-            result = [res[m_count] for res in results]
+            result = results[m_count]
             if not isinstance(m, CountsMP):
                 # We don't need to cast to arrays when using qml.counts. qml.math.array is not viable
                 # as it assumes all elements of the input are of builtin python types and not belonging
                 # to any particular interface
-                result = qml.math.stack(result, like=interface)
+                result = qml.math.array(result, like=interface)
```

Comment:
I originally changed this to `qml.math.stack` because `qml.math.array` fails when trying to create an array using a list/tuple of other arrays. There is an unrelated issue with the tests currently, so it's hard to tell if this is actually causing any trouble or not. Just bringing it up here in case any interface tests for `dynamic_one_shot` start failing.

Reply:
The stacking should already have happened in `processing_fn` now, so hopefully this problem doesn't reappear.

Reply:
Looks like the tests are still passing. Nice!